src/cargo/ops/cargo_compile.rs

```diff
@@ -474,6 +474,8 @@ fn scrape_target_config(config: &Config, triple: &str)
                     let list = value.list(&k)?;
                     output.cfgs.extend(list.iter().map(|v| v.0.clone()));
                 }
+                // FIXME: what happens with "warning" and "rerun-if-changed",
+                // should they really go into metadata?
```

Comment:
It still feels wrong that `warning` and `rerun-if-changed` are treated as user-defined metadata in build script overrides, but not in build scripts themselves. They should either be ignored or reported as errors.
I'll submit a separate PR fixing this later.